src/inapp/utils.ts

```diff
@@ -232,6 +232,48 @@ const mediaQueryLg = global?.matchMedia?.(
 );
 const mediaQueryXl = global?.matchMedia?.('(min-width: 1301px)');
 
+/**
+ *
+ * @returns { HTMLIFrameElement } iframe with sandbox and hidden styling applied for of an inapp message to render with
+ */
+const generateSecuredIFrame = () => {
+  const iframe = document.createElement('iframe');
+  iframe.setAttribute('id', 'iterable-iframe');
+  iframe.setAttribute('sandbox', 'allow-same-origin allow-popups');
+  /*
+    _display: none_ would remove the ability to set event handlers on elements
+    so instead we choose to hide it visibly with CSS but not actually remove
+    its interact-ability
+
+    https://snook.ca/archives/html_and_css/hiding-content-for-accessibility
+  */
+  iframe.style.cssText = `
+    position: fixed !important;
+    top: 0;
+    left: 0;
+    height: 1px;
+    width: 1px;
+    overflow: hidden;
+  `;
+
+  return iframe;
+};
+
+/**
+ *
+ * @param html
+ * @returns { string } html string of an inapp message to wrap in an iframe for render
+ */
+export const wrapWithIFrame = (html: string): string => {
+  const iframe = generateSecuredIFrame();
+  return iframe.outerHTML.replace(
```

Comment:
You should be able to write to the iframe's document even before it's attached to the DOM, is that not reliable in all supported browsers?

Reply:
When I was trying to write to the document it didn't exist so it wouldn't write, do you know which attribute I should be writing to? I tried both contentWindow.document and contentDocument but they were null until it was attached to the DOM

Reply:
i've been struggling with this technique. Stringifying code is often a source of vulnerability, but in this case I'm not seeing anything obviously.

That said, I also don't know that it's necessary to write the onload event as a string. Is it possible instead to use the form:

```
iframe.onload = function() { iframe.contentWindow.document.write(html); }
```

Is that something the team tried already? I am a little unsure regarding whether you get the closures given the iframe boundary, but it should be a little cleaner if it works. We'd want to see how IE/Edge handle it even if it works in Chrome.

Reply:
So this function is just a transform function to take the raw HTML string and place it inside an iframe through the onload event. It hasn't reached the customer's app code yet so they still expect it as a string from the API response to render into their app. Since it's still technically in the API layer I don't think we want to mess with the response type and it should match our actual API response type which is just a string for the HTML which is why I've done it this way.